pkgs/data/fonts/openmoji/default.nix

```diff
@@ -0,0 +1,79 @@
+{ stdenv
+, fetchFromGitHub
+, scfbuild
+, variant ? "color" # "color" or "black"
+}:
+
+let
+  filename = builtins.replaceStrings
+    [ "color"              "black"              ]
+    [ "OpenMoji-Color.ttf" "OpenMoji-Black.ttf" ]
+    variant;
+
+in stdenv.mkDerivation rec {
+  pname = "openmoji";
+  version = "12.0.0";
+
+  src = fetchFromGitHub {
+    owner = "hfg-gmuend";
+    repo = pname;
+    rev = version;
+    sha256 = "0h8sr74gbdmcm8vv8pxlig50mmbf9w9lh4p8ih2lqqi5iry1dvhj";
+  };
+
+  nativeBuildInputs = [
+    scfbuild
+  ];
+
+  # Some id and data-* attributes in the svg files contain unicode characters,
+  # which scfbuild (a python2 program) does not like
+  # (https://github.com/13rac1/scfbuild/issues/14).
+  # Fortunately, it's only metadata which we can remove:
+  postPatch = ''
+    sed -Ei 's/(id|data-[^=]*)="[^"]*"//g' black/svg/*.svg color/svg/*.svg
+  '';
+
+  buildPhase = ''
+    # Bash reimplementation of helpers/export-svg-font.js
+    # so we don't need to build all the node deps first
+    hexcodes=()
+    missingGlyphBlack='./black/svg/25A1.svg'
+    missingGlyphColor='./color/svg/25A1.svg'
+    for f in ./color/svg/*.svg; do
+      basename=$(basename "$f" .svg)
+      hexcodes+=(''${basename//-/ })
+      filename=$(basename "$f");
+      cp "./color/svg/$filename" "./font/tmp-color/$filename"
+      cp "./black/svg/$filename" "./font/tmp-black/$filename"
+    done
+
+    hexcodes=($(uniq<<<"$hexcodes"))
```

Comment:
This bash reimplementation is broken, unfortunately, but happily it's an easy fix for the issue I noticed.  Here's the resulting commit: 0a5b480de54b71f95eefb77fbbaf795a6734df53 (on my "fork").

This may not be enough, as the resulting color variant isn't color and I'm not sure all the expected glyphs end up included.  But I can imagine reasons my tree is to blame-- do the resulting files work for you? What have you tried them with?
(grabbing the font files provided for the release don't seem to work for me either, which also suggests it's something on my end...)

I had a fear that hexcodes actually needed to include the range of values between the hyphenated pair but offhand I'm not sure whether that's what the javascript this is based on actually does O:).

----

Perhaps we should simply fetch and use the provided generated font files? (assuming they work for folks?)

Reply:
Oops, yeah, thanks for the fix, I'll cherry pick your commit shortly.

The weird thing is that both your version and mine do work for me. Probably because that part only handles missing glyphs.

I tested it on Firefox, one of the few programs that support color fonts. Install the font then go to about:config and set `font.name-list.emoji` to `OpenMoji`. Here's the result:

![2019-10-02-152635_957x419_scrot](https://user-images.githubusercontent.com/8182846/66051877-2809f980-e51f-11e9-93f1-9bd4fae74111.png)

A bit glitchy on some glyphs, but it works!

---

I think the hypenated pairs mean that the emoji is for a ligature between the two glyphs. For example 🇮 (REGIONAL INDICATOR SYMBOL LETTER I) + 🇹 (REGIONAL INDICATOR SYMBOL LETTER T) = 🇮🇹 (italian flag) (try to copy this text and delete the space between the two letters: 🇮 🇹)

Reply:
Cherry-picked